main.py

```diff
@@ -93,24 +180,31 @@ def should_nag(message: discord.Message) -> bool:
     3. Embed is safe and there is a video
     """
     if not re.search(r"(//|mobile\.)(twitter|x)\.com", message.content):
-        return False
+        return None
     if not message.embeds:
-        return True
+        return NagType.FULL
     # sensitive check _must_ be done before other types
     # to prevent mixing NSFW and SFW links from embedding
     if any(_is_sensitive_tweet_embed(em) for em in message.embeds):
         channel = message.channel
         if channel.type in (discord.ChannelType.private, discord.ChannelType.group):
             # always allow NSFW in (group) DMs
-            return True
-        if hasattr(channel, "nsfw"):
+            return NagType.FULL
+        if hasattr(channel, "nsfw") and getattr(channel, "nsfw") is True:
             # .nsfw seems to not exist on some channel types
             # and discord.py type narrowing is limited
-            # so this will have to do
-            return getattr(channel, "nsfw") is True
+            return NagType.FULL
+        # nsfw in sfw channel, prevent embed
+        print("Skipping NSFW embed in SFW channel")
+        return None
     if any(_is_video_tweet(em) for em in message.embeds):
-        return True
-    return False
+        return NagType.VIDEO
+    # remaining case is POTENTIALLY multi-image, check fx API
+    tweets: list[tuple[str, str]] = twitter_url_regex.findall(message.content)
+    for user, tid in tweets:
+        # if there's at least one, return the first nag type
+        return get_fx_nagtype(user, tid)
```

Comment:
Would it not make more sense, if multiple tweets in the message have different "priorities", to pick the highest commonly acceptable one? In a trivial case, a message with two tweet links, the first of which is broken, would result in _no_ response, even if the second one should get a response (say, a mosaic)

Reply:
Updated logic to match